packages/trader/src/AppV2/Components/CurrentSpot/current-spot-display.tsx

```diff
@@ -0,0 +1,133 @@
+    INC: 'increment',
+    DEC: 'decrement',
+    ADD10: 'add10',
+} as const;
+const TOTAL_ANIMATION_TIME = 240;
+
+const CurrentSpotDisplay = ({ has_tick_count, spot, tick }: TCurrentSpotDisplayProps) => {
+    const last_digit = Number(spot?.slice(-1));
+    const prev_spot = usePrevious(spot);
+    const prev_last_digit = Number(prev_spot?.slice(-1));
+
+    const [should_enter_from_top, setShouldEnterFromTop] = React.useState(false);
+    const [should_enter_from_bottom, setShouldEnterFromBottom] = React.useState(false);
+    const [next_displayed_last_digit, setNextDisplayedLastDigit] = React.useState(last_digit);
+    const [displayed_last_digit, setDisplayedLastDigit] = React.useState(last_digit);
+
+    const last_digit_ref = React.useRef(prev_last_digit);
+    const spin_interval_id = React.useRef<ReturnType<typeof setInterval>>();
+    const spin_timeout_id = React.useRef<ReturnType<typeof setTimeout>>();
+    const spinning_wrapper_ref = React.useRef<HTMLDivElement>(null);
+
+    const spinLastDigit = (
+        action: typeof ACTIONS[keyof typeof ACTIONS],
+        interval_ms: number,
+        start: number,
+        end: number
+    ) => {
+        clearInterval(spin_interval_id.current);
+        const interval_id = setInterval(() => {
+            if (action === ACTIONS.INC && last_digit_ref.current < end) {
+                last_digit_ref.current = (last_digit_ref.current + 1) % 10;
+            } else if (action === ACTIONS.DEC && last_digit_ref.current > end) {
+                last_digit_ref.current = Math.abs(last_digit_ref.current - 1) % 10;
+            } else if (action === ACTIONS.ADD10 && last_digit_ref.current < start + 10) {
+                last_digit_ref.current += 1;
+            } else if (
+                action === ACTIONS.ADD10 ? last_digit_ref.current === start + 10 : last_digit_ref.current === end
+            ) {
+                clearInterval(interval_id);
+            }
+            setNextDisplayedLastDigit(last_digit_ref.current % 10);
+        }, interval_ms);
+        spin_interval_id.current = interval_id;
+    };
+
+    React.useEffect(() => {
+        if (isNaN(prev_last_digit) || isNaN(last_digit) || !spot) return;
+
+        const diff = Math.abs(Number(prev_last_digit) - last_digit);
+        const timeout_speed = diff > 0 ? Math.floor(TOTAL_ANIMATION_TIME / diff) : TOTAL_ANIMATION_TIME;
+        const should_increment = Number(prev_last_digit) <= last_digit;
+
+        spinning_wrapper_ref.current?.style.setProperty('--animation-time', `${timeout_speed}ms`);
+
+        if (should_increment) {
+            setShouldEnterFromTop(true);
+        } else {
+            setShouldEnterFromBottom(true);
+        }
+
+        spin_timeout_id.current = setTimeout(() => {
+            setShouldEnterFromTop(false);
+            setShouldEnterFromBottom(false);
+            setDisplayedLastDigit(last_digit_ref.current % 10);
+        }, TOTAL_ANIMATION_TIME); // equal to total animation time
```

Comment:
maybe we dont need the comment

Reply:
right, since we introduced the constant, will remove it )